python/python/lance/dataset.py

```diff
@@ -242,15 +272,15 @@ def scanner(
         batch_size: Optional[int] = None,
         batch_readahead: Optional[int] = None,
         fragment_readahead: Optional[int] = None,
-        scan_in_order: bool = True,
+        scan_in_order: bool = None,
         fragments: Optional[Iterable[LanceFragment]] = None,
         full_text_query: Optional[Union[str, dict]] = None,
         *,
-        prefilter: bool = False,
-        with_row_id: bool = False,
-        with_row_address: bool = False,
-        use_stats: bool = True,
-        fast_search: bool = False,
+        prefilter: bool = None,
+        with_row_id: bool = None,
+        with_row_address: bool = None,
+        use_stats: bool = None,
+        fast_search: bool = None,
```

Comment:
These changes in defaults should not be breaking changes since the defaults in `ScannerBuilder` match the defaults that used to be here.

By using `None` we can easily tell if the user is specifying a non-default value, in which case we will override whatever is in `default_scan_options`.